src/Appwrite/Auth/OAuth2/Linkedin.php

```diff
@@ -159,12 +158,12 @@ public function getUserName(string $accessToken): string
         $user = $this->getUser($accessToken);
         $name = '';
 
-        if (isset($user['localizedFirstName'])) {
-            $name = $user['localizedFirstName'];
+        if (isset($user['given_name'])) {
+            $name = $user['given_name'];
         }
 
-        if (isset($user['localizedLastName'])) {
-            $name = (empty($name)) ? $user['localizedLastName'] : $name . ' ' . $user['localizedLastName'];
+        if (isset($user['family_name'])) {
+            $name = (empty($name)) ? $user['family_name'] : $name . ' ' . $user['family_name'];
```

Comment:
Let's just use the name from LinkedIn:

```php
        if (isset($user['name'])) {
            return $user['name'];
        }
```

Reply:
in some cases name might be not present and given_name or family_name might be present so should i add this case also with the other two cases? like if name is not present take info from given_name or family_name ?

Reply:
Oh really? That's odd behavior...sure, go ahead and fallback to given and family name.